hr_employee_firstname/tests/test_hr_employee_firstname.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2014 Savoir-faire Linux. All Rights Reserved.
+# Copyright 2016-2019 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import odoo
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+from odoo.tools import submap
+
+
+class TestEmployeeFirstname(TransactionCase):
+    def setUp(self):
+        super().setUp()
+
+        # Create 3 employees to concatenate the firstname and lastname
+        # in name_related
+        self.employee1_id = self.env["hr.employee"].create(
+            {"firstname": "Jan", "lastname": "Van-Eyck"}
+        )
+        self.employee2_id = self.env["hr.employee"].create(
+            {"firstname": "Jean-Pierre", "lastname": "Carnaud"}
+        )
+        self.employee3_id = self.env["hr.employee"].create(
+            {"firstname": "Famke", "lastname": "Jenssens"}
+        )
+
+        # Create 3 employees for split the name_related to
+        # firstname and lastname
+        self.employee10_id = self.env["hr.employee"].create({"name": " Jan Van-Eyck"})
+        self.employee20_id = self.env["hr.employee"].create(
+            {"name": "Jean-Pierre Carnaud"}
+        )
+        self.employee30_id = self.env["hr.employee"].create({"name": "JenssensFamke"})
+
+    def test_get_name(self):
+        """
+        Validate the _get_name method is concatenating
+        the firstname and lastname
+        """
+        # Check for employee1
+        self.assertEqual(self.employee1_id.name, "Jan Van-Eyck")
+
+        # Check for employee2
+        self.assertEqual(self.employee2_id.name, "Jean-Pierre Carnaud")
+
+        # Check for employee3
+        self.assertEqual(self.employee3_id.name, "Famke Jenssens")
+
+    def test_onchange(self):
+        """
+        Validate the get_name method is not failing
+        """
+        field_onchange = self.env["hr.employee"].new({})._get_fields_spec()
```

Comment:
This is not an onchange anymore AFAIK, so I think you shouldn't neither name it as so nor using `new()`.